### DISCUSSION TITLE
Creating a `Column` with a Vector of Strings

### DISCUSSION THREAD

**Discussion by aadityadhruv** [hecrj/iced#2022] [Q&A] [answered]
I am still new to Rust, so I am not sure if this stems from my lack of understanding of Rust or Iced. I am trying to learn Iced & Rust, and am experimenting with the library. I have a `Vec<String>` which I would like to display as a column. However, looking at the `column` function implementation, the children parameter has type `Vec<Element<'_, Message, Renderer>, Global>`. Firstly, I am not too sure I understand what this signature means. I understand that there is a generic type in the Element part, but what are the `Message`, `Renderer` and `Global` parts? 

Secondly, how would I go about making a column with a vector of strings? I could `iter()` and `map` it into some Element but I am not sure how that works or what that means.

**Comment by lufte** [comment, marked as answer]
You will need to create a `Text` widget for each `String` in you collection and then you can either use `with_children` to create the `Column` with all its children in one go or `push` to add items one by one.

```rust
    fn view(&self) -> iced::Element<Message> {
        let mut window = iced::widget::Column::with_children(vec![
            iced::widget::Text::new("hello").into(),
            iced::widget::Text::new("world").into(),
        ]);
        window = window.push(iced::widget::Text::new("!"));
        return window.into();
    }
```

Notice how `with_children` requires a vector of `Elements`, so I call `into()` on my widgets, but `push` only requires an `Into<Element>` so I can pass the widget directly.

There are also some macros and helper functions that make the whole thing look simpler, although I'm just finding out about them now.

```rust
    fn view(&self) -> iced::Element<Message> {
        let mut window = iced::widget::column![
            iced::widget::text("hello"),
            iced::widget::text("world"),
        ];
        window = window.push(iced::widget::text("!"));
        return window.into();
    }
```
